File: cli-rust/crates/oa-cmd-onboard/src/auth/credentials.rs
/// Credential storage and default model reference constants.
///
/// Provides functions to persist API keys and OAuth credentials to disk,
/// along with default model reference constants for each provider.
///
/// Source: `src/commands/onboard-auth.credentials.ts`

/// Default model reference for ZAI.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `ZAI_DEFAULT_MODEL_REF`
pub const ZAI_DEFAULT_MODEL_REF: &str = "zai/glm-4.7";

/// Default model reference for Xiaomi.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `XIAOMI_DEFAULT_MODEL_REF`
pub const XIAOMI_DEFAULT_MODEL_REF: &str = "xiaomi/mimo-v2-flash";

/// Default model reference for OpenRouter.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `OPENROUTER_DEFAULT_MODEL_REF`
pub const OPENROUTER_DEFAULT_MODEL_REF: &str = "openrouter/auto";

/// Default model reference for Vercel AI Gateway.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `VERCEL_AI_GATEWAY_DEFAULT_MODEL_REF`
pub const VERCEL_AI_GATEWAY_DEFAULT_MODEL_REF: &str =
    "vercel-ai-gateway/anthropic/claude-opus-4.6";

/// Default model reference for Cloudflare AI Gateway.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - re-exported from cloudflare module
pub const CLOUDFLARE_AI_GATEWAY_DEFAULT_MODEL_REF: &str =
    "cloudflare-ai-gateway/claude-opus-4-6";

/// Default model reference for xAI (Grok).
///
/// Source: `src/commands/onboard-auth.credentials.ts` - re-exported from models
pub const XAI_DEFAULT_MODEL_REF: &str = "xai/grok-4";

/// Credential type for auth profile storage.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - credential types
#[derive(Debug, Clone)]
pub enum CredentialType {
    /// API key credential.
    ApiKey {
        /// Provider identifier.
        provider: String,
        /// The API key.
        key: String,
    },
    /// OAuth credential.
    OAuth {
        /// Provider identifier.
        provider: String,
        /// OAuth tokens and metadata.
        email: Option<String>,
    },
}

/// Parameters for upserting an auth profile.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `upsertAuthProfile` params
#[derive(Debug, Clone)]
pub struct UpsertAuthProfileParams {
    /// Profile ID (e.g., "anthropic:default").
    pub profile_id: String,
    /// The credential to store.
    pub credential: CredentialType,
    /// Agent directory to store credentials in.
    pub agent_dir: Option<String>,
}

/// Store an Anthropic API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setAnthropicApiKey`
pub fn build_anthropic_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "anthropic:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "anthropic".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a Gemini API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setGeminiApiKey`
pub fn build_gemini_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "google:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "google".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a Moonshot API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setMoonshotApiKey`
pub fn build_moonshot_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "moonshot:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "moonshot".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a xAI API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setXaiApiKey`
pub fn build_xai_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "xai:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "xai".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store an OpenRouter API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setOpenrouterApiKey`
pub fn build_openrouter_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "openrouter:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "openrouter".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a ZAI API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setZaiApiKey`
pub fn build_zai_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "zai:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "zai".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a Venice API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setVeniceApiKey`
pub fn build_venice_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "venice:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "venice".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

/// Store a Qianfan API key credential.
///
/// Source: `src/commands/onboard-auth.credentials.ts` - `setQianfanApiKey`
pub fn build_qianfan_api_key_profile(key: &str) -> UpsertAuthProfileParams {
    UpsertAuthProfileParams {
        profile_id: "qianfan:default".to_string(),
        credential: CredentialType::ApiKey {
            provider: "qianfan".to_string(),
            key: key.to_string(),
        },
        agent_dir: None,
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn build_anthropic_profile() {
        let profile = build_anthropic_api_key_profile("sk-test");
        assert_eq!(profile.profile_id, "anthropic:default");
        match profile.credential {
            CredentialType::ApiKey { provider, key } => {
                assert_eq!(provider, "anthropic");
                assert_eq!(key, "sk-test");
            }
            _ => panic!("expected ApiKey credential"),
        }
    }

    #[test]
    fn build_xai_profile() {
        let profile = build_xai_api_key_profile("xai-key");
        assert_eq!(profile.profile_id, "xai:default");
    }

    #[test]
    fn model_refs_format() {
        assert_eq!(ZAI_DEFAULT_MODEL_REF, "zai/glm-4.7");
        assert_eq!(XIAOMI_DEFAULT_MODEL_REF, "xiaomi/mimo-v2-flash");
        assert_eq!(OPENROUTER_DEFAULT_MODEL_REF, "openrouter/auto");
    }
}

File: backend/internal/goproviders/onboard/credentials.go
// onboard/credentials.go — 凭证管理（API Key 设置 + OAuth 凭证写入）
// 对应 TS 文件: src/commands/onboard-auth.credentials.ts
// 包含所有 set*ApiKey 函数和 writeOAuthCredentials 函数。
package onboard

import (
	"fmt"
	"os"
	"path/filepath"
	"regexp"
	"strings"

	"github.com/Acosmi/ClawAcosmi/internal/goproviders/authprofile"
	"github.com/Acosmi/ClawAcosmi/internal/goproviders/types"
)

// ──────────────────────────────────────────────
// 默认模型引用常量
// ──────────────────────────────────────────────

// ZaiDefaultModelRef Z.AI 默认模型引用。
const ZaiDefaultModelRef = "zai/glm-5"

// XiaomiDefaultModelRef 小米默认模型引用。
const XiaomiDefaultModelRef = "xiaomi/mimo-v2-flash"

// OpenrouterDefaultModelRef OpenRouter 默认模型引用。
const OpenrouterDefaultModelRef = "openrouter/auto"

// HuggingfaceDefaultModelRef Hugging Face 默认模型引用。
const HuggingfaceDefaultModelRef = "huggingface/deepseek-ai/DeepSeek-R1"

// TogetherDefaultModelRef Together 默认模型引用。
const TogetherDefaultModelRef = "together/moonshotai/Kimi-K2.5"

// LitellmDefaultModelRef LiteLLM 默认模型引用。
const LitellmDefaultModelRef = "litellm/claude-opus-4-6"

// VercelAiGatewayDefaultModelRef Vercel AI Gateway 默认模型引用。
const VercelAiGatewayDefaultModelRef = "vercel-ai-gateway/anthropic/claude-opus-4.6"

// CloudflareAiGatewayDefaultModelRef Cloudflare AI Gateway 默认模型引用。
const CloudflareAiGatewayDefaultModelRef = "cloudflare-ai-gateway/anthropic/claude-sonnet-4-20250514"

// KilocodeDefaultModelRef Kilocode 默认模型引用。
var KilocodeDefaultModelRef = "kilocode/" + KilocodeDefaultModelID

// ──────────────────────────────────────────────
// 环境变量引用模式
// ──────────────────────────────────────────────

var envRefPattern = regexp.MustCompile(`^\$\{([A-Z][A-Z0-9_]*)\}$`)

// ApiKeyStorageOptions API 密钥存储选项。
type ApiKeyStorageOptions struct {
	SecretInputMode types.SecretInputMode
}

// ──────────────────────────────────────────────
// 密钥引用构建函数
// ──────────────────────────────────────────────

func buildEnvSecretRef(id string) types.SecretRef {
	return types.SecretRef{
		Source:   types.SecretRefSourceEnv,
		Provider: types.DefaultSecretProviderAlias,
		ID:       id,
	}
}

func parseEnvSecretRef(value string) *types.SecretRef {
	matches := envRefPattern.FindStringSubmatch(value)
	if matches == nil {
		return nil
	}
	ref := buildEnvSecretRef(matches[1])
	return &ref
}

// ProviderEnvVars 提供者对应的环境变量名称映射。
var ProviderEnvVars = map[string][]string{
	"anthropic":             {"ANTHROPIC_API_KEY"},
	"openai":                {"OPENAI_API_KEY"},
	"google":                {"GOOGLE_API_KEY", "GEMINI_API_KEY"},
	"minimax":               {"MINIMAX_API_KEY"},
	"moonshot":              {"MOONSHOT_API_KEY"},
	"kimi-coding":           {"KIMI_CODING_API_KEY"},
	"volcengine":            {"VOLCENGINE_API_KEY", "ARK_API_KEY"},
	"byteplus":              {"BYTEPLUS_API_KEY"},
	"synthetic":             {"SYNTHETIC_API_KEY"},
	"venice":                {"VENICE_API_KEY"},
	"zai":                   {"ZAI_API_KEY", "ZHIPU_API_KEY"},
	"xiaomi":                {"XIAOMI_API_KEY"},
	"openrouter":            {"OPENROUTER_API_KEY"},
	"kilocode":              {"KILOCODE_API_KEY"},
	"litellm":               {"LITELLM_API_KEY"},
	"vercel-ai-gateway":     {"VERCEL_AI_GATEWAY_API_KEY"},
	"cloudflare-ai-gateway": {"CLOUDFLARE_AI_GATEWAY_API_KEY"},
	"opencode":              {"OPENCODE_API_KEY"},
	"together":              {"TOGETHER_API_KEY"},
	"huggingface":           {"HUGGINGFACE_API_KEY", "HF_TOKEN"},
	"qianfan":               {"QIANFAN_API_KEY"},
	"xai":                   {"XAI_API_KEY"},
	"mistral":               {"MISTRAL_API_KEY"},
}

func resolveProviderDefaultEnvSecretRef(provider string) (types.SecretRef, error) {
	envVars, ok := ProviderEnvVars[provider]
	if !ok || len(envVars) == 0 {
		return types.SecretRef{}, fmt.Errorf("Provider \"%s\" 没有默认环境变量映射（secret-input-mode=ref）", provider)
	}
	for _, v := range envVars {
		if strings.TrimSpace(v) != "" {
			return buildEnvSecretRef(v), nil
		}
	}
	return types.SecretRef{}, fmt.Errorf("Provider \"%s\" 没有默认环境变量映射（secret-input-mode=ref）", provider)
}

// resolveApiKeySecretInput 解析 API Key 输入为最终存储形式。
func resolveApiKeySecretInput(provider string, input string, opts *ApiKeyStorageOptions) (string, *types.SecretRef) {
	// 尝试环境变量引用解析
	envRef := parseEnvSecretRef(input)
	if envRef != nil {
		return "", envRef
	}
	// ref 模式
	if opts != nil && opts.SecretInputMode == types.SecretInputModeRef {
		ref, err := resolveProviderDefaultEnvSecretRef(provider)
		if err == nil {
			return "", &ref
		}
	}
	// 明文
	return strings.TrimSpace(input), nil
}

// ApiKeyCredential API Key 凭证结构。
type ApiKeyCredential struct {
	Type     string            `json:"type"`
	Provider string            `json:"provider"`
	Key      string            `json:"key,omitempty"`
	KeyRef   *types.SecretRef  `json:"keyRef,omitempty"`
	Metadata map[string]string `json:"metadata,omitempty"`
}

func buildApiKeyCredential(provider string, input string, metadata map[string]string, opts *ApiKeyStorageOptions) ApiKeyCredential {
	key, ref := resolveApiKeySecretInput(provider, input, opts)
	cred := ApiKeyCredential{
		Type:     "api_key",
		Provider: provider,
	}
	if ref != nil {
		cred.KeyRef = ref
	} else {
		cred.Key = key
	}
	if metadata != nil {
		cred.Metadata = metadata
	}
	return cred
}

// ──────────────────────────────────────────────
// UpsertAuthProfile 占位（调用 authprofile 包）
// 窗口 8 将替换为真实实现
// ──────────────────────────────────────────────

func upsertAuthProfileCredential(profileID string, credential interface{}, agentDir string) {
	store := authprofile.EnsureAuthProfileStore(agentDir, nil)
	if store == nil {
		return
	}
	credMap, ok := credential.(map[string]interface{})
	if !ok {
		// ApiKeyCredential 等结构体转 map
		if apiKeyCred, ok := credential.(ApiKeyCredential); ok {
			credMap = map[string]interface{}{
				"type":     apiKeyCred.Type,
				"provider": apiKeyCred.Provider,
			}
			if apiKeyCred.Key != "" {
				credMap["key"] = apiKeyCred.Key
			}
			if apiKeyCred.KeyRef != nil {
				credMap["keyRef"] = apiKeyCred.KeyRef
			}
			if apiKeyCred.Metadata != nil {
				credMap["metadata"] = apiKeyCred.Metadata
			}
		} else {
			return
		}
	}
	authprofile.UpsertAuthProfile(store, profileID, credMap)
	_ = authprofile.SaveAuthProfileStore(store, agentDir)
}

// resolveAuthAgentDir 解析认证代理目录。
func resolveAuthAgentDir(agentDir string) string {
	if agentDir != "" {
		return agentDir
	}
	// 默认使用 ~/.openacosmi/agent 目录
	home, _ := os.UserHomeDir()
	return filepath.Join(home, ".openacosmi", "agent")
}

// ──────────────────────────────────────────────
// WriteOAuthCredentials 写入 OAuth 凭证
// ──────────────────────────────────────────────

// WriteOAuthCredentialsOptions OAuth 凭证写入选项。
type WriteOAuthCredentialsOptions struct {
	SyncSiblingAgents bool
}

// safeRealpath 安全解析真实路径。
func safeRealpath(dir string) string {
	resolved, err := filepath.EvalSymlinks(filepath.Clean(dir))
	if err != nil {
		return ""
	}
	return resolved
}

// resolveSiblingAgentDirs 解析同级代理目录。
func resolveSiblingAgentDirs(primaryAgentDir string) []string {
	normalized, _ := filepath.Abs(primaryAgentDir)
	parentOfAgent := filepath.Dir(normalized)
	candidateAgentsRoot := filepath.Dir(parentOfAgent)
	looksLikeStandardLayout := filepath.Base(normalized) == "agent" &&
		filepath.Base(candidateAgentsRoot) == "agents"

	var agentsRoot string
	if looksLikeStandardLayout {
		agentsRoot = candidateAgentsRoot
	} else {
		home, _ := os.UserHomeDir()
		agentsRoot = filepath.Join(home, ".openacosmi", "state", "agents")
	}

	entries, err := os.ReadDir(agentsRoot)
	if err != nil {
		entries = nil
	}
	var discovered []string
	for _, entry := range entries {
		if entry.IsDir() || entry.Type()&os.ModeSymlink != 0 {
			discovered = append(discovered, filepath.Join(agentsRoot, entry.Name(), "agent"))
		}
	}

	seen := make(map[string]bool)
	var result []string
	allDirs := append([]string{normalized}, discovered...)
	for _, dir := range allDirs {
		real := safeRealpath(dir)
		if real != "" && !seen[real] {
			seen[real] = true
			result = append(result, real)
		}
	}
	return result
}

// WriteOAuthCredentials 写入 OAuth 凭证到认证存储。
// 对应 TS: writeOAuthCredentials()
func WriteOAuthCredentials(
	provider string,
	creds map[string]interface{},
	agentDir string,
	options *WriteOAuthCredentialsOptions,
) (string, error) {
	email := "default"
	if e, ok := creds["email"].(string); ok && strings.TrimSpace(e) != "" {
		email = strings.TrimSpace(e)
	}
	profileID := provider + ":" + email
	resolvedAgentDir, _ := filepath.Abs(resolveAuthAgentDir(agentDir))

	credential := make(map[string]interface{})
	credential["type"] = "oauth"
	credential["provider"] = provider
	for k, v := range creds {
		credential[k] = v
	}

	// 主写入
	upsertAuthProfileCredential(profileID, credential, resolvedAgentDir)

	// 同级同步（尽力而为）
	if options != nil && options.SyncSiblingAgents {
		targetDirs := resolveSiblingAgentDirs(resolvedAgentDir)
		primaryReal := safeRealpath(resolvedAgentDir)
		for _, targetDir := range targetDirs {
			targetReal := safeRealpath(targetDir)
			if targetReal != "" && primaryReal != "" && targetReal == primaryReal {
				continue
			}
			upsertAuthProfileCredential(profileID, credential, targetDir)
		}
	}
	return profileID, nil
}

// ──────────────────────────────────────────────
// set*ApiKey 函数
// ──────────────────────────────────────────────

// SetAnthropicApiKey 设置 Anthropic API 密钥。
func SetAnthropicApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("anthropic:default",
		buildApiKeyCredential("anthropic", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetOpenaiApiKey 设置 OpenAI API 密钥。
func SetOpenaiApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("openai:default",
		buildApiKeyCredential("openai", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetGeminiApiKey 设置 Gemini API 密钥。
func SetGeminiApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("google:default",
		buildApiKeyCredential("google", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetMinimaxApiKey 设置 MiniMax API 密钥。
func SetMinimaxApiKey(key string, agentDir string, profileID string, opts *ApiKeyStorageOptions) {
	if profileID == "" {
		profileID = "minimax:default"
	}
	provider := "minimax"
	if parts := strings.SplitN(profileID, ":", 2); len(parts) > 0 && parts[0] != "" {
		provider = parts[0]
	}
	upsertAuthProfileCredential(profileID,
		buildApiKeyCredential(provider, key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetMoonshotApiKey 设置 Moonshot API 密钥。
func SetMoonshotApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("moonshot:default",
		buildApiKeyCredential("moonshot", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetKimiCodingApiKey 设置 Kimi Coding API 密钥。
func SetKimiCodingApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("kimi-coding:default",
		buildApiKeyCredential("kimi-coding", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetVolcengineApiKey 设置火山引擎 API 密钥。
func SetVolcengineApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("volcengine:default",
		buildApiKeyCredential("volcengine", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetByteplusApiKey 设置 BytePlus API 密钥。
func SetByteplusApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("byteplus:default",
		buildApiKeyCredential("byteplus", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetSyntheticApiKey 设置 Synthetic API 密钥。
func SetSyntheticApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("synthetic:default",
		buildApiKeyCredential("synthetic", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetVeniceApiKey 设置 Venice API 密钥。
func SetVeniceApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("venice:default",
		buildApiKeyCredential("venice", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetZaiApiKey 设置 Z.AI API 密钥。
func SetZaiApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("zai:default",
		buildApiKeyCredential("zai", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetXiaomiApiKey 设置小米 API 密钥。
func SetXiaomiApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("xiaomi:default",
		buildApiKeyCredential("xiaomi", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetOpenrouterApiKey 设置 OpenRouter API 密钥。
func SetOpenrouterApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	safeKey := key
	if safeKey == "undefined" {
		safeKey = ""
	}
	upsertAuthProfileCredential("openrouter:default",
		buildApiKeyCredential("openrouter", safeKey, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetCloudflareAiGatewayConfig 设置 Cloudflare AI Gateway 配置。
func SetCloudflareAiGatewayConfig(accountID, gatewayID, apiKey string, agentDir string, opts *ApiKeyStorageOptions) {
	normalizedAccountID := strings.TrimSpace(accountID)
	normalizedGatewayID := strings.TrimSpace(gatewayID)
	metadata := map[string]string{
		"accountId": normalizedAccountID,
		"gatewayId": normalizedGatewayID,
	}
	upsertAuthProfileCredential("cloudflare-ai-gateway:default",
		buildApiKeyCredential("cloudflare-ai-gateway", apiKey, metadata, opts), resolveAuthAgentDir(agentDir))
}

// SetLitellmApiKey 设置 LiteLLM API 密钥。
func SetLitellmApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("litellm:default",
		buildApiKeyCredential("litellm", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetVercelAiGatewayApiKey 设置 Vercel AI Gateway API 密钥。
func SetVercelAiGatewayApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("vercel-ai-gateway:default",
		buildApiKeyCredential("vercel-ai-gateway", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetOpencodeZenApiKey 设置 OpenCode Zen API 密钥。
func SetOpencodeZenApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("opencode:default",
		buildApiKeyCredential("opencode", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetTogetherApiKey 设置 Together API 密钥。
func SetTogetherApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("together:default",
		buildApiKeyCredential("together", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetHuggingfaceApiKey 设置 Hugging Face API 密钥。
func SetHuggingfaceApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("huggingface:default",
		buildApiKeyCredential("huggingface", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetQianfanApiKey 设置千帆 API 密钥。
func SetQianfanApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("qianfan:default",
		buildApiKeyCredential("qianfan", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetXaiApiKey 设置 xAI API 密钥。
func SetXaiApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("xai:default",
		buildApiKeyCredential("xai", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetMistralApiKey 设置 Mistral API 密钥。
func SetMistralApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("mistral:default",
		buildApiKeyCredential("mistral", key, nil, opts), resolveAuthAgentDir(agentDir))
}

// SetKilocodeApiKey 设置 Kilocode API 密钥。
func SetKilocodeApiKey(key string, agentDir string, opts *ApiKeyStorageOptions) {
	upsertAuthProfileCredential("kilocode:default",
		buildApiKeyCredential("kilocode", key, nil, opts), resolveAuthAgentDir(agentDir))
}
